sqlglot/generator.py

```diff
@@ -3795,20 +3795,20 @@ def credentials_sql(self, expression: exp.Credentials) -> str:
         if isinstance(cred_expr, exp.Literal):
             # Redshift case: CREDENTIALS <string>
             credentials = self.sql(expression, "credentials")
-            credentials = f"CREDENTIALS {credentials}"
+            credentials = f" CREDENTIALS {credentials}" if credentials else ""
```

Comment:
I don't like this pattern, we should generally only add whitespace in the generators of parent nodes because otherwise if you were to generate a `Credentials` on its own for whatever reason you'd end up with a leading space. Can we return back the spacing in `copy_sql`?